src/sidebar/components/annotation-omega.js

```diff
@@ -62,9 +66,19 @@ function AnnotationOmega({
     createDraft(annotation, { ...draft, text });
   };
 
+  const onSave = () => {
+    // TODO flash errors if not authenticated or if annotation is empty?
+    setIsSaving(true);
+    annotationsService
+      .save(annotation)
+      .catch(err => flash.error(err.message, 'Saving annotation failed'))
+      .finally(() => {
+        setIsSaving(false);
+      });
```

Comment:
We don't currently load a polyfill for `Promise.prototype.finally` for browsers which have `Promise` but not with this method. See the example of how we [test for](https://github.com/hypothesis/client/blob/e0f8835ca18b7d12e988c7329b95bd9ef7730d58/src/shared/polyfills/index.js#L53) and then [load a polyfill](https://github.com/hypothesis/client/blob/e0f8835ca18b7d12e988c7329b95bd9ef7730d58/src/shared/polyfills/es2016.js#L1) for `Array.prototype.includes`(from ES2016) for an example of how to implement this.

Reply:
Want to be sure I'm parsing this correctly: Are you saying the task here is to load a polyfill for `Promise.prototype.finally` using our established conventions for feature testing/loading polyfills?

Reply:
...Thinking on this a little further, is there any sort of automated sanity check we could use for situations like this (i.e. if Babel notices we're using something we haven't polyfilled)? Perhaps that's tricky...

Reply:
> Are you saying the task here is to load a polyfill for Promise.prototype.finally using our established conventions for feature testing/loading polyfills?

Yes.

> Thinking on this a little further, is there any sort of automated sanity check we could use for situations like this (i.e. if Babel notices we're using something we haven't polyfilled)? 

Hmm, I'll have a think. Babel has limited type information so there is a limit to what it can do. A type checker (eg. TypeScript looking at our JSDoc comments) might be able to do a better job. Or we could set up our test runner to disable any "new" functions which are not whitelisted as having been polyfilled ... that might be the easiest to do.